modules/tinymce/src/core/main/ts/undo/Operations.ts

```diff
@@ -165,3 +205,20 @@ export const ignore = (locks: Locks, callback: () => void): void => {
     locks.set(locks.get() - 1);
   }
 };
+
+const ignoreAsync = (locks: Locks, callback: () => void | Promise<void>): void | Promise<void> => {
+  try {
+    locks.set(locks.get() + 1);
+    const result = callback();
+
+    if (result) {
+      locks.set(locks.get() + 1);
+
+      return result.finally(() => {
```

Comment:
The callback should only return `void` or `Promise<void>`, so if it is not void it should be a promise.

Reply:
Yes but use the isPromiseLike check instead of a truthy check. The callback is user defined so types will not be save here if they return true this will throw but will be catches by the finally not sure that is the intent.

Reply:
They'd be breaking the API to do that, but sure, I'll add it if you think it's prudent.